scenes/game/local_game.gd

```diff
@@ -1666,12 +1692,29 @@ class Player:
 		amount = min(amount, movement_limit)
 
 		var previous_location = arena_location
-		var other_location = parent._get_player(parent.get_other_player(my_id)).arena_location
+		var other_player = parent._get_player(parent.get_other_player(my_id))
+		var other_location = other_player.arena_location
 		var new_location
 		if arena_location < other_location:
 			new_location = min(other_location-1, arena_location+amount)
 		else:
 			new_location = max(other_location+1, arena_location-amount)
+
```

Comment:
I'm starting to worry about the amount of code we're duplicating between close/advance/retreat.
Maybe close/retreat can basically be the same function just in different directions?
And close is really just advance but opponent is blocking you.

I'll let you decide if you want to do this or not.

Reply:
I know for Kokonoe I was also adding some stop_on_buddy functionality, which is slightly different from your blocked_by_buddy since it is inclusive.

Reply:
i wonder, is there anything stopping us from having advance/retreat/close just redirect to move_to (with a blocking option for close)?

Reply:
Well, you don't know exactly where you're moving to depending on the opponent for advance.
I think the most reasonable is like:
move_in_direction_by_amount(direction : int, amount : int, blocked_by_opponent : bool)
Then it moves you in that direction 1 at a time, checking blocked_by_opponent, blocked_by_buddy_id stuff, and the new stop_on_buddy thing I'm adding.
When you land on the opponent (and you weren't blocked by them), you move a bonus space in direction.

How does that sound?

Reply:
Close/Advance/Retreat will then calculate the direction first and pass that in.

Reply:
Probably more simple than what I was trying out; I'll see how well it works